### PR TITLE
Update storage sync generated client code and add tests

### PR DESCRIPTION
**Pull request by pahnallan**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/storage/Azure.ResourceManager.Storage/tests/Helpers/StorageTestBase.cs [hunk 1]
```diff
@@ -11,6 +11,8 @@
 using NUnit.Framework;
 using Azure.Core;
 using Azure.ResourceManager.Models;
+using Castle.DynamicProxy;
```
Comment (anpint) [inline sdk/storage/Azure.ResourceManager.Storage/tests/Helpers/StorageTestBase.cs:14]: Looks like you're changing a file in the storage SDK. Any particular reason? I think we need to revert the changes to this file.

Comment (pahnallan) [reply sdk/storage/Azure.ResourceManager.Storage/tests/Helpers/StorageTestBase.cs:14]: This one accidentally slipped in, reverting the change now

Comment (pahnallan) [reply sdk/storage/Azure.ResourceManager.Storage/tests/Helpers/StorageTestBase.cs]: Just reverted it back to normal

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Azure.ResourceManager.StorageSync.Tests.csproj [hunk 2]
```diff
@@ -2,4 +2,24 @@
...
+    </Compile>
+  </ItemGroup>
+  <ItemGroup>
+    <None Update="Properties\Settings.settings">
+      <Generator>SettingsSingleFileGenerator</Generator>
+      <LastGenOutput>Settings.Designer.cs</LastGenOutput>
+    </None>
+  </ItemGroup>
+  <ItemGroup>
+    <Folder Include="SessionRecords\CloudEndpointTests\" />
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Azure.ResourceManager.StorageSync.Tests.csproj:19]: Why do we need include in csproj?

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/CustomPolicy/StorageSyncResponseNullFilterPolicy.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace Azure.ResourceManager.StorageSync.Tests.CustomPolicy
+{
+    public class StorageSyncResponseNullFilterPolicy : HttpPipelineSynchronousPolicy
+    {
+        public override void OnReceivedResponse(HttpMessage message)
+        {
+            //The delete method response only contains "value":null, after filter, nothing would left
+            if (message.Request.Method != RequestMethod.Delete && message.Response.ContentStream != null)
+            {
+                StreamReader reader = new StreamReader(message.Response.ContentStream);
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/CustomPolicy/StorageSyncResponseNullFilterPolicy.cs:19]: 1. use var
2. use "using"

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/CustomPolicy/StorageSyncResponseNullFilterPolicy.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+using System.IO;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+
+namespace Azure.ResourceManager.StorageSync.Tests.CustomPolicy
+{
+    public class StorageSyncResponseNullFilterPolicy : HttpPipelineSynchronousPolicy
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/CustomPolicy/StorageSyncResponseNullFilterPolicy.cs:12]: Would you please add some documentation

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/StorageSyncManagementTestBase.cs [hunk 5]
```diff
@@ -37,5 +60,21 @@ protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionReso
             var lro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
             return lro.Value;
         }
+
+        public async Task<ResourceGroupResource> CreateResourceGroupAsync()
+        {
+            string resourceGroupName = Recording.GenerateAssetName("resTrack2");
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/StorageSyncManagementTestBase.cs:66]: afs-sdk-rg

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+        private string _storageSyncServiceName;
+        private string _syncGroupName;
+        private string _cloudEndpointName;
+        private StorageSyncServiceCreateOrUpdateContent _storageSyncServiceCreateOrUpdateContent;
+        private StorageSyncGroupCreateOrUpdateContent _storageSyncGroupCreateOrUpdateContent;
+        private CloudEndpointCreateOrUpdateContent _cloudEndpointCreateOrUpdateContent;
+        private StorageSyncServiceResource _storageSyncServiceResource;
+        private StorageSyncGroupResource _storageSyncGroupResource;
+
+        public CloudEndpointTests(bool async) : base(async) //, RecordedTestMode.Record)
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs:27]: remove

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+        {
+        }
+
+        [SetUp]
+        public async Task CreateStorageSyncResources()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            _storageSyncServiceName = Recording.GenerateAssetName("sss-cepcreate");
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs:35]: afs-sdk-sss-create

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+            _storageSyncGroupResource = (await _storageSyncServiceResource.GetStorageSyncGroups().CreateOrUpdateAsync(WaitUntil.Completed, _syncGroupName, _storageSyncGroupCreateOrUpdateContent)).Value;
+            Assert.NotNull(_storageSyncGroupResource);
+            StorageSyncManagementTestUtilities.VerifySyncGroupProperties(_storageSyncGroupResource, true);
+        }
+
+        [TearDown]
+        public async Task DeleteStorageSyncResources()
+        {
+            var storageSyncServiseExists = (await _resourceGroup.GetStorageSyncServices().ExistsAsync(_storageSyncServiceName)).Value;
+            if (storageSyncServiseExists)
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs:58]: storageSyncServiceExists

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+            Assert.NotNull(cloudEndpointResource);
+            StorageSyncManagementTestUtilities.VerifyCloudEndpointProperties(cloudEndpointResource, true);
+        }
+
+        [Test]
+        [RecordedTest]
+        public async Task CloudEndpointGetTest()
+        {
+            // Create CloudEndpoint
+            CloudEndpointResource cloudEndpointResource = (await _storageSyncGroupResource.GetCloudEndpoints().CreateOrUpdateAsync(WaitUntil.Completed, _cloudEndpointName, _cloudEndpointCreateOrUpdateContent)).Value;
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs:90]: Option 1 :  Get as is, Check HasValue using Assert.  Use .Value
Option 2 : Use GetValueOrDefault()

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs [hunk 10]
```diff
@@ -0,0 +1,167 @@
...
+
+        [SetUp]
+        public async Task CreateStorageSyncResources()
+        {
+            _resourceGroup = await CreateResourceGroupAsync();
+            _storageSyncServiceName = Recording.GenerateAssetName("sss-cepcreate");
+            _syncGroupName = Recording.GenerateAssetName("sg-cepcreate");
+            _cloudEndpointName = Recording.GenerateAssetName("cepcreate");
+
+            _storageSyncServiceCreateOrUpdateContent = StorageSyncManagementTestUtilities.GetDefaultStorageSyncServiceParameters();
```
Comment (ankushbindlish2) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Tests/CloudEndpointTests.cs:39]: Why are they not abstracted out in common test class setup - whatever make sense

**Review by ankushbindlish2 — CHANGES_REQUESTED** [review]
I have left few changes to incorporate

File: sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Azure.ResourceManager.StorageSync.Tests.csproj (hunk 2 above)
Comment (pahnallan) [reply sdk/storagesync/Azure.ResourceManager.StorageSync/tests/Azure.ResourceManager.StorageSync.Tests.csproj]: I believe these were automatically added when running the initial test recording, but I removed them in the latest commit and it should be resolved now.

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs [hunk 11]
```diff
@@ -380,6 +382,12 @@ namespace Azure.ResourceManager.StorageSync.Models
         public static bool operator !=(Azure.ResourceManager.StorageSync.Models.ChangeDetectionMode left, Azure.ResourceManager.StorageSync.Models.ChangeDetectionMode right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class CloudEndpointAfsShareMetadataCertificatePublicKeys
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:385]: This name is weird, not sure if we can have a better name

File: sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md [hunk 12]
```diff
@@ -8,8 +8,8 @@ azure-arm: true
 csharp: true
 library-name: StorageSync
 namespace: Azure.ResourceManager.StorageSync
-require: https://github.com/Azure/azure-rest-api-specs/blob/2614c5038cb384e864c1608e76be61a5ee84cb02/specification/storagesync/resource-manager/readme.md
-tag: package-2020-09-01
+require: https://github.com/Azure/azure-rest-api-specs/blob/d1eee5499dbf9281debdc90c4f4cbc7470fb8d6d/specification/storagesync/resource-manager/readme.md
```
Comment (ArthurMa1978) [inline sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md:11]: Please update the CHANGELOG.md and give the date of when do you want to release the new package

File: sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs (hunk 11 above)
Comment (pahnallan) [reply sdk/storagesync/Azure.ResourceManager.StorageSync/api/Azure.ResourceManager.StorageSync.netstandard2.0.cs:385]: This name comes from another api (AfsShareMetadataCertificatePublicKeys API) and was mostly put in swagger for compliance. It won't be used by customers so it might not be worth it to change.

File: sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md (hunk 12 above)
Comment (pahnallan) [reply sdk/storagesync/Azure.ResourceManager.StorageSync/src/autorest.md:11]: Sounds good, just updated the CHANGELOG.md. 

Could we get this released at the next earliest date? I slotted the release for 2022-10-21 for now but it can be changed depending on your availability for publishing the release. Thanks!

**Review by ArthurMa1978 — APPROVED** [review]

**Review by ArthurMa1978 — APPROVED** [review]
